packages/rum/src/rum.ts

```diff
@@ -259,127 +225,26 @@ export function startRumEventCollection(
-        return
-      }
-      primaryBatch.add(message)
-      if (replicaBatch) {
-        replicaBatch.add(withReplicaApplicationId(message))
-      }
-    },
-    stop: () => {
-      stopped = true
-    },
-    upsert: (message: Context, key: string) => {
-      if (stopped) {
-        return
-      }
-      primaryBatch.upsert(message, key)
-      if (replicaBatch) {
-        replicaBatch.upsert(withReplicaApplicationId(message), key)
-      }
-    },
-  }
-}
-
-interface AssembleWithoutAction {
-  view: ViewContext
-  rum: RumContext
-}
-
-interface AssembleWithAction extends AssembleWithoutAction {
-  action?: ActionContext
-}
-
-type RumEventHandler = <T extends RawRumEvent>(
-  assemble: (event: T, { view, action, rum }: AssembleWithAction) => RumEvent,
-  callback: (message: Context, event: RumEvent) => void
-) => (startTime: number, event: T, customerContext?: Context) => void
-
-function makeRumEventHandler(
-  parentContexts: ParentContexts,
-  session: RumSession,
-  rumContextProvider: () => RumContext,
-  globalContextProvider: () => Context
-): RumEventHandler {
-  return function rumEventHandler<T extends RawRumEvent>(
-    assemble: (event: T, { view, action, rum }: AssembleWithAction) => RumEvent,
-    callback: (message: Context, event: RumEvent) => void
-  ) {
-    return (startTime: number, event: T, savedGlobalContext?: Context, customerContext?: Context) => {
-      const view = parentContexts.findView(startTime)
-      if (session.isTracked() && view && view.sessionId) {
-        const action = parentContexts.findAction(startTime)
-        const rumEvent = assemble(event, { action, view, rum: rumContextProvider() })
-        const message = combine(
-          savedGlobalContext || globalContextProvider(),
-          customerContext,
-          withSnakeCaseKeys(rumEvent)
-        )
-        callback(message, rumEvent)
-      }
-    }
-  }
-}
-
-interface BrowserWindow extends Window {
-  _DATADOG_SYNTHETICS_BROWSER?: unknown
-}
-
-function getSessionType() {
-  return (window as BrowserWindow)._DATADOG_SYNTHETICS_BROWSER === undefined ? SessionType.USER : SessionType.SYNTHETICS
-}
-
-export function trackRumEvents(lifeCycle: LifeCycle, session: RumSession, handler: RumEventHandler, batch: RumBatch) {
-  const assembleWithoutAction = (event: RumViewEvent | RumUserActionEvent, { view, rum }: AssembleWithoutAction) =>
-    combine(rum, view, event)
-  const assembleWithAction = (
-    event: RumErrorEvent | RumResourceEvent | RumLongTaskEvent,
-    { view, action, rum }: AssembleWithAction
-  ) => combine(rum, view, action, event)
+export function trackRumEvents(lifeCycle: LifeCycle, session: RumSession) {
+  const handler = (
+    startTime: number,
+    rawRumEvent: RawRumEvent,
+    savedGlobalContext?: Context,
+    customerContext?: Context
+  ) =>
+    lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, {
+      customerContext,
+      rawRumEvent,
+      savedGlobalContext,
+      startTime,
+    })
```

Comment:
Maybe each `track*` functions could use `lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, { ... })` directly? I think it could be even easier to understand, at the cost of being a bit more verbose. Ex:


```ts
function trackCustomUserAction(lifeCycle: LifeCycle) {
  lifeCycle.subscribe(
    LifeCycleEventType.CUSTOM_ACTION_COLLECTED,
    ({ action: { name, type, context: customerContext, startTime }, context: savedGlobalContext }) => {
      lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, {
        startTime,
        rawRumEvent: {
          date: getTimestamp(startTime),
          evt: {
            name,
            category: RumEventCategory.USER_ACTION,
          },
          userAction: {
            type,
          },
        },
        savedGlobalContext,
        customerContext
      })
    }
  )
}
```

Reply:
Since: 
- these track functions are planned to be refactored soon
- updating these functions will impact unit tests not related to the scope of this PR

I'd be in favor of keeping it as is for now and think about it when we get there.

Reply:
:+1: